helix-core/src/main/java/org/apache/helix/model/ParticipantHistory.java

```diff
@@ -64,6 +69,20 @@ public ParticipantHistory(ZNRecord znRecord) {
     super(znRecord);
   }
 
+  /**
+   * @return The list field for HISTORY
+   */
+  public List<String> getHistory() {
+    return _record.getListField(ConfigProperty.HISTORY.name());
+  }
+
+  /**
+   * @return The list field for OFFLINE
+   */
+  public List<String> getOffline() {
+    return _record.getListField(ConfigProperty.OFFLINE.name());
+  }
+
```

Comment:
I think you don't need these 2 methods to be public?

Reply:
They are public because 
1. we should provide accessors since now there are actual code usage of ParticipantHistory; and, 
2. an integration test calls these functions.

Reply:
1. you have the methods which return the timestamps. The raw string is hard to use and we shall not return it.
2. integration test can rely on protected methods.

Reply:
I disagree with this. For a HelixProperty class, if there is a need of a field, there should be an accessor. It is true that we have timestamp parsers now, but for example, HISTORY contains other informations too. Down the road, there may be other code that relies on other HISTORY information. What would happen then? An accessor will be built. Right now there is a need to these fields, and I don't see a reason to not create accessors. 

For integration test it's not in the same package, so I don't think package private (nor protected) can help. This is a minor point; there are workarounds. My main point is the previous paragraph.

Reply:
"What would happen then?" Then we add other methods to access this information explicitly. The string return which requires additional parse is a bad idea in general.
1. What if we change the string format? These methods return string will be non-backward-compatible. And we don't want to restrict ourselves unnecessarily, right?
2. Like you add the getOnlineTimestamp now, we can add other methods that parse the HISTORY string and return the required information.
3. The necessity of testing does not require us to make it public. We can make the test in the same package. Or read the field directly from ZNRecord. There are many ways to achieve it and public is the most impactful and costly (maintenance cost) way. I think it is not good option.